extensions-contrib/opentelemetry-extensions/src/main/java/org/apache/druid/data/input/opentelemetry/protobuf/OpenTelemetryMetricsProtobufReader.java

```diff
@@ -0,0 +1,183 @@
+  private final String metricDimension;
+  private final String valueDimension;
+  private final String metricAttributePrefix;
+  private final String resourceAttributePrefix;
+  private final DimensionsSpec dimensionsSpec;
+
+  public OpenTelemetryMetricsProtobufReader(
+      DimensionsSpec dimensionsSpec,
+      ByteEntity source,
+      String metricDimension,
+      String valueDimension,
+      String metricAttributePrefix,
+      String resourceAttributePrefix
+  )
+  {
+    this.dimensionsSpec = dimensionsSpec;
+    this.source = source;
+    this.metricDimension = metricDimension;
+    this.valueDimension = valueDimension;
+    this.metricAttributePrefix = metricAttributePrefix;
+    this.resourceAttributePrefix = resourceAttributePrefix;
+  }
+
+  @Override
+  public CloseableIterator<InputRow> read()
+  {
+    return CloseableIterators.withEmptyBaggage(readAsList().iterator());
+  }
+
+  List<InputRow> readAsList()
+  {
+    try {
+      return parseMetricsData(MetricsData.parseFrom(source.getBuffer()));
+    }
+    catch (InvalidProtocolBufferException e) {
+      throw new ParseException(e, "Protobuf message could not be parsed");
+    }
+  }
+
+  private List<InputRow> parseMetricsData(final MetricsData metricsData)
+  {
+    return metricsData.getResourceMetricsList()
+        .stream()
+        .flatMap(resourceMetrics -> {
+          Map<String, Object> resourceAttributes = resourceMetrics.getResource()
+              .getAttributesList()
+              .stream()
+              .collect(Collectors.toMap(kv -> resourceAttributePrefix + kv.getKey(),
+                  kv -> getStringValue(kv.getValue())));
+          return resourceMetrics.getInstrumentationLibraryMetricsList()
+              .stream()
+              .flatMap(libraryMetrics -> libraryMetrics.getMetricsList()
+                  .stream()
+                  .flatMap(metric -> parseMetric(metric, resourceAttributes).stream()));
+        })
+        .collect(Collectors.toList());
+  }
+
+  private List<InputRow> parseMetric(Metric metric, Map<String, Object> resourceAttributes)
+  {
+    final List<InputRow> inputRows;
+    String metricName = metric.getName();
+    switch (metric.getDataCase()) {
+      case SUM: {
+        inputRows = new ArrayList<>(metric.getSum().getDataPointsCount());
+        metric.getSum()
+            .getDataPointsList()
+            .forEach(dataPoint -> inputRows.add(parseNumberDataPoint(dataPoint, resourceAttributes, metricName)));
+        break;
+      }
+      case GAUGE: {
+        inputRows = new ArrayList<>(metric.getGauge().getDataPointsCount());
+        metric.getGauge()
+            .getDataPointsList()
+            .forEach(dataPoint -> inputRows.add(parseNumberDataPoint(dataPoint, resourceAttributes, metricName)));
+        break;
+      }
+      // TODO Support HISTOGRAM and SUMMARY metrics
+      default:
+        throw new IllegalStateException("Unexpected value: " + metric.getDataCase());
```

Comment:
@marcusgreer I just realized we can't just throw an exception for histogram metrics, since this will blow up for some of the metrics we have, which do have some histogram component. We should instead ignore those fields for now, so that we don't throw away the entire record.